File: java/src/main/java/io/fivexer/sdk/model/RevertedWeights.java
package io.fivexer.sdk.model;

import java.util.List;

/**
 * Which workers had their learned routing weights restored.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class RevertedWeights {
    private List<String> reverted;
    private Integer count;

    public List<String> getReverted() { return reverted; }
    public Integer getCount() { return count; }
}

File: php/src/Model/RevertedWeights.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * Which workers had their learned routing weights restored.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class RevertedWeights
{
    public function __construct(
        /** @var list<string> */
        public readonly array $reverted,
        public readonly int $count,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            reverted: \array_values(\array_map(strval(...), $data['reverted'] ?? [])),
            count: (int) ($data['count'] ?? 0),
        );
    }
}
